src/btree/bt_cursor.c

```diff
@@ -978,20 +1084,31 @@ __cursor_truncate(WT_SESSION_IMPL *session,
 	 * instantiated the end cursor, so we know that page is pinned in memory
 	 * and we can proceed without concern.
 	 */
-retry:	WT_RET(__wt_btcur_remove(start));
+retry:	WT_RET(__wt_btcur_search(start));
+
+	/*
+	 * XXX KEITH
+	 * When the btree cursor code sets/clears the cursor flags (rather than
+	 * the cursor layer), the set/clear goes away, only the assert remains.
+	 */
+	F_CLR((WT_CURSOR *)start, WT_CURSTD_KEY_SET);
+	F_SET((WT_CURSOR *)start, WT_CURSTD_KEY_INT);
+	WT_ASSERT(session,
+	    F_MASK((WT_CURSOR *)start, WT_CURSTD_KEY_SET) == WT_CURSTD_KEY_INT);
 
 	/*
 	 * Reset ret each time through so that we don't loop forever in
 	 * the cursor equals case.
 	 */
 	for (ret = 0;;) {
+		if ((ret = rmfunc(session, start, 1)) != 0)
```

Comment:
As discussed on the phone - I would be more comfortable if this code wasn't reordered.